src/ppx/string_util.cpp

```diff
@@ -68,5 +69,34 @@ std::optional<std::pair<std::string_view, std::string_view>> SplitInTwo(std::str
     return std::make_pair(firstSubstring, secondSubstring);
 }
 
+std::string WrapText(const std::string& s, size_t width, size_t indent = 0)
+{
+    if (indent >= width) {
+        return s;
+    }
+    auto remainingString = TrimCopy(s);
+
+    size_t      textWidth   = width - indent;
+    std::string wrappedText = "";
+    while (remainingString != "") {
+        size_t lineLength = remainingString.find_last_of(" \t", textWidth);
+        if (lineLength == std::string::npos) {
+            lineLength = textWidth;
+        }
+
+        std::string newLine = remainingString.substr(0, lineLength);
+        TrimRight(newLine);
+
+        wrappedText += std::string(indent, ' ') + newLine + "\n";
+        if (lineLength >= remainingString.length()) {
```

Comment:
I am curious, do you need this check or can you let it break when it reaches the while condition above?

Reply:
Ah right, I think line 84 should probably be
lineLength = std::max(textWidth, remainingString.length());
and then you wouldn't need this check and wouldn't rely on the substr above being fine with more characters requested than exist

Reply:
yes, right! Thanks!